pkg/controller/direct/bigquerydataset/dataset_controller.go

```diff
@@ -253,15 +269,16 @@ func (a *Adapter) Update(ctx context.Context, updateOp *directbase.UpdateOperati
 	if desired.Access == nil || len(desired.Access) == 0 {
 		resource.Access = a.actual.Access
 	}
-	updateDatasetCall := a.gcpService.Datasets.Update(parent.ProjectID, datasetId, resource)
-	updated, err := updateDatasetCall.Do()
+	datasetMetadataToUpdate := BigQueryDatasetMetadata_ToMetadataToUpdate(mapCtx, resource)
```

Comment:
Not sure if it is possible/easier to convert directly from `krm.BigQueryDatasetSpec` to `pb.DatasetMetadataToUpdate` in "Update". I feel it might be easier to read and maintain in the future.

Reply:
This update does not support `updateMask`, so I would still need to go through all the non-immutable fields and update the clone of the `a.actual` object.

I could convert the `desired` object directly from `krm.BigQueryDatasetSpec` to `pb.DatasetMetadataToUpdate`. But I think the problem is that `a.actual` is of type `pb.DatasetMetadata`. After I updated the `a.actual` object with new values from the `desired` object, I would still need to convert it to `pb.DatasetMetadataToUpdate`.